Justin Payne
============


> A scientific toolmaker building pathogen biosurveillence networks.
>
> Specialties: Integration, Full-stack Web, Bioinformatics Service Architectures.

---

(I) Information
---------------
1155 Ripley St Apt 1814  
Silver Spring, MD 20910  

*Justin.Payne@fda.hhs.gov*  
+1 402.909.1436  

GitHub Username: [crashfrog](http://github.com/crashfrog)

---

(II) Education
--------------
Date              | Degree and Program
-----------------:|:------------------
``2009 - 2012``   |Earned **Bachelor of Science in Biochemistry**, minors in Chemistry and Mathematics, University of Nebraska - Lincoln
``1999 - 2003``   |Studied for Bachelors of Arts in English and Computer Science, Gustavus Adolphus College (on hiatus)
``Aug 1998``      |Certificate of French proficiency, Université Paris-Sorbonne


---

(III) Professional Experience
-----------------------------
Date                     | Organization and Duties
:-----------------------:|:-----------------------
``Oct 2012 - Apr 2015``     |**FDA-CFSAN-ORS-DM-MMSB (ORAU Fellowship)**
``Apr 2015 - Aug 2016``     |**FDA-CFSAN-ORS-DM-MMSB (GS-403-11)**
``Aug 2016 - Current``      |**FDA-CFSAN-OAO-DB (GS-403-12)**
                            |**Bioinformatician in Microbiology** supporting the **analysis and interpretation of molecular data from next generation DNA sequencing technologies,** and the **development and maintenance of whole-genome pathogen surveillence networks** to aid in the detection, identification, and characterization of microbial food-borne pathogens. **Primary requirements analyst, author, and maintainer** of the FDA-hosted codebase and informatic infrastructure for FDA/State joint GenomeTrakr Network, FDA/CDC Real-Time Listeria Genomic Surveillance Project, and FDA-CFSAN/MN/WA/NY Real-Time Salmonella Enteriditis Project, resulting in the deposit of more than 15,000 bacterial genomes into publicly-accessible databases. Provided bioinformatics system development consultation on an informal basis to researchers at diverse US and international agencies and groups. Demonstrated expertise in command-line operation of Linux and Mac OS X (desktop, server, and high-performance clusters.) Author of high-performance analysis tools in **C/C++, Python, Java**, and JavaScript to **analyze and interpret molecular data from DNA sequencing platforms: Illumina HiSeq/MiSeq/NextSeq, Pacific Biosciences RSII, Ion Torrent PGM, Roche 454, Oxford Nanopore MinION.** Developed GenomicsDB database for isolate metadata, sequencing scheduling, and high-throughput submission to national biotechnology databases such as NCBI GenBank and NCBI SRA, and primary data architect in the project to migrate this system to Genohm's SLims Genomic Information Management System (GIMS). **Wrote integration-layer packages in Python to automatically collect, assess, quality-control, assemble, and analyze genomic sequencing data from Illumina MiSeq/NextSeq, IonTorrent PGM, and PacBio RS sequencing instruments** locally or connected to the FDA network via VPN connections or through Illumina BaseSpace cloud services.  Developed sra-quick-submit, an easy-to-use NCBI Sequence Read Archive (SRA) submission tool used by researchers around the world to submit their own genomes to the GenomeTrakr project. Utilized High-Performance Computing (HPC) and Field-Programmable Gate Array (FPGA) coprocessor platforms for high-throughput genomic science. **Evaluated over 40 bioinformatics tools and methods for pipeline incorporation and analysis use, determined their relevance to the detection and typing of foodborne pathogens** and to the food safety mission as a whole, and provided local support for their deployment and use. **Assisted in developing SOP for conducting next-generation genomic analysis of foodborne pathogens.** Developer of Platter, a virtual WebDAV filesystem that exposes sequencing data to bioinformatics researchers on multiple semantic paths to enable an elegant and exploratory environment for computational biology.
                         |
``Feb 2012 - Oct 2012``  |**Trojan Defense, LLC**
                         |**Analysis Software Programmer** and primary **Internet of Things architect** developing **distributed nuclear materials monitoring architectures** based on proprietary solid-state radiologic sensor. Specialized in "bare-metal" C/C++ on Atmel-family embedded SoC (Systems on a Chip.) Developed custom evaluation boards based on **Arduino-family system development boards** (Uno rev3, Mini, Mega, Leo, Yun.) Used professional illustration tools (Adobe family, Inkscape) to produce technical illustrations, patent documentation, and marketing materials. Wrote proposals for Small Business Innovation Research (SBIR) grants and RFP responses for DTRA, DoD, DoE, and other Federal agencies.
                         |
``May 2005 - Dec 2006``  |**University of Missouri-Columbia/USDA-ARS**
                         |**Biological Field Research Aide (GS-1)** assisting in field studies in agricultural entomology, jointly as an appointment to the USDA-ARS and as an employee of the University of Missouri, Columbia. Developed professional web presence and custom content management system (CMS) using Apache and PHP. Developed laboratory setup and tools for electroanntennaographic bioassay of _Diabrotica vergifera v._ (western corn rootworm) and _Loxosceles reclusa_ (brown recluse) species for detection of host identification factors. Constructed field study equipment using basic and power carpentry tools. Managed small (3-6 person) groups of fellow assistants. Worked independently to ensure adherence to study protocols.
                         |
``Sept 1998 - Jan 2003`` |**Gustavus Adolphus College, Dept. of Information Technologies**
                         |**Tech Support Consultant** providing phone and site-based technical support on Windows 95/98/ME/2000/NT/XP, Mac OS 9/10, and Linux (Debian/Ubuntu/Fedora/Slackware) systems for students, staff, and faculty. Wrote training materials.

---



(IV) Technologies
-----------------

**Platforms**

* Linux (Ubuntu/CentOS/RHEL/Arch)
* Mac OS X (desktop, server)
* Atmel family of embedded microprocessors
* Xilinx Spartan-6 FPGAs
* Amazon EC2

**Programming Languages**

* Python
* Perl
* PHP
* C/C++
* Java
* JavaScript
* HTML5+CSS3

**Databases**

* MySQL
* SQL Server 2008
* PostgreSQL
* Oracle
* Neo4J

**Development Frameworks**

* Django
* Flask
* AWT/Swing
* Angular.js
* React.js
* JSPlumb
* Gunicorn

---

(V) Publications
----------------

2. Marc Allard, Tim Muruvanda, Errol Strain, Ruth Timme, Yan Luo, Charles Wang, Christine Keys, **Justin Payne**, Tony Cooper, Khai Luong, Yi Song, Chen-Shan Chin, Jonas Korlach, Richard Roberts, Peter Evans, Steven Musser, and Eric Brown. *A fully assembled genome sequence for Salmonella. Javiana strain CFSAN001992.* Genome Announc. 2013 Mar 21;1(2):e0008113.

1. Narjol González-Escalona, Melinda McFarland, Lydia Rump, **Justin Payne**, Denis Andrzejewski, Eric Brown, Peter Evans, and Timothy Crowley. *Draft genome sequences of two O104:H21 Escherichia coli isolates causing hemorrhagic colitis during an outbreak in Montana in 1994 provide insight into their pathogenicity.* Genome Announc. 2013 Oct 3;1(5). pii: e00805-13.

3. Maria Hoffmann, Tim Muruvanda, Marc Allard, Jonas Korlach, Richard Roberts, Ruth Timme, **Justin Payne**, Patrick McDermott, Peter Evans, Jianghong Meng, Eric Brown, and Shaohua Zhao. *Complete Genome Sequence of a Multidrug-Resistant Salmonella enterica Serovar Typhimurium var. 5- Strain Isolated from Chicken Breast.* Genome Announc. 2014 Apr 3;2(2). pii: e00294-14

4. Julie Haendiges, Ruth Timme, Marc Allard, Robert Myers, **Justin Payne**, Eric Brown, Peter Evans, and Narjol González-Escalona. *Draft genomes of clinical Vibrio parahaemolyticus strains isolated in Maryland (2010-2013).* Genome Announc. 2014 Aug 7;2(4). pii: e00776-14.

21. Muruvanda, T., Pirone, C., Hoffmann, M., Allard, M. W0., Wang, C., Strain, E., Timme, R., Luo, Y., Keys, E. C., **Payne, J.**, Luong, K., Song, Y., Chin, C., Korlach, J., Roberts, R. J., Musser, S. M., Evans, P. & Brown, E. W. *New discoveries in Salmonella genomes closure.* 114th American Society for Microbiology Annual Meeting, Boston, MA, USA, (May, 2014). AND 4th Annual FDA Foods & Vet Medicine Science and Research Conference, White Oak, MD. July, 2014. AND Next-generation sequencing (NGS) technology, data formats standardization and promotion of interoperability protocols, Bethesda, MD, USA (September, 2014)

22. Allard, M. W., Pirone, C., Muruvanda, T., Hoffman, M., Soler-García, A., Wang, C., Strain, E., Timme, R., **Payne, J.**, Luo, Y., Keys, C. E., Ferrieira, C., Chin, C., Korlach, J., Musser, S. M., Zhao, S., Stones, R., Roberts, R. J., Evans, P., and Brown, E. W. *New discoveries from closing Salmonella genomes using Pacific Biosciences continuous long reads.* 15th Advances in Genome Biology and Technology Meeting (ABGT), Marco Island, FL, USA (2014). AND 4th Annual FDA Foods & Vet Medicine Science and Research Conference, White Oak, MD. July, 2014.

5. Maria Hoffmann, Tim Muruvanda, Cary Pirone, Jonas Korlach, Ruth Timme, **Justin Payne**, Peter Evans, Jianghong Meng, Eric Brown, and Marc Allard. *First Fully Assembled Genome Sequence of Salmonella enterica Serovar Cubana associated with a Multistate Outbreak in the United States.* Genome Announc. October, 2014.

6. Cary Pirone-Davies, Maria Hoffmann, Richard J. Roberts, Tim Muruvanda, Ruth Timme, Errol Strain, Yan Luo, **Justin Payne**, Khai Luong, Yi Song, Yu-Chih Tsai, Matthew Boitano, Tyson Clark, Jonas Korlach, Peter Evans, Marc Allard. *Genome-wide methylation patterns in Salmonella enterica subsp. Enterica serovars.* PLoS ONE 10(4): e0123639. doi:10.1371/journal.pone.0123639

8. **Justin Payne**, Erica Lindroth, Kate Kneeland, Steven R. Skoda, Fatima Mustafa, Muhammad Irfan Ullah, John E. Foster. *Bootsie - Estimation of Coefficient of Variation of AFLP Data by Bootstrap Analysis.* Pak. Entomol. 36(2):79-81

7. Guojie Cao, Marc Allard, Maria Hoffmann, Steven Monday, Tim Muruvanda, Yan Luo, **Justin Payne**, Lydia Rump, Kevin Meng, Shaohua Zhao, Patrick McDermott, Eric Brown, and Jianghong Meng. *Complete Sequences of Six IncA/C Plasmids of Multidrug-Resistant Salmonella Newport.* Genome Announc. February, 2015.

9. Maria Hofmann, **Justin Payne**, Richard Roberts, Marc Allard, Eric Brown, and James Pettengill. *Complete Genome Sequence of Salmonella enterica Serovar Agona 460004 2-1  Associated with a Multi-State Outbreak in the United States.* Genome Announc. July, 2015.

10. Hua Wang, Yi Chen, Sherry Ayers, David Melka, Anna Maounounen-Laasri, **Justin Payne**, Jie Zheng, Insook Son, Ruth Timme, George Kastanis, Thomas Hammack, Errol Strain, Marc Allard, Peter Evans, and Eric Brown. *Draft Genome Sequence of Salmonella enterica subsp. enterica Serovar Give Isolated from an Imported Chili Powder Product.* Genome Announc. July, 2015.

11. Emily Pettengill, Maria Hoffman, Rachel Binet, Richard J. Roberts, **Justin Payne**, Marc Allard, Valeria Michelacci, Fabio Minelli and Stefano Morabito. *Complete Genome Sequence of Enteroinvasive **Escherichia coli** O96:H19 Associated with a Severe Food-borne Outbreak.* Genome Announc. August, 2015.

12. Steve Davis, James B Pettengill, Yan Luo, **Justin Payne**, Al Shpuntoff, Hugh Rand, and Errol Strain. *CFSAN SNP Pipeline: an automated method for constructing SNP matricies from next-generation sequence data.* [PeerJ Computer Science 1:e20](https://dx.doi.org/10.7717/peerj-cs.20), 2015.

13. Kuan Yao, Tim Muruvanda, Richard Roberts, **Justin Payne**, Marc Allard, and Maria Hoffman. *Complete Genome and Methylome Sequences of two Salmonella enterica species.* Genome Announc. January, 2016.

20. Kuan Yao, Tim Muruvanda, Richard Roberts, **Justin Payne**, Marc Allard, and Maria Hoffmann. *Complete Genome and Methylome Sequences of Salmonella Panama ATCC 7378 and Salmonella Sloterdijk ATCC 15791.* Genome Announc. Feburary, 2016.

82. Hua Wang, Jie Zheng, Sherry Ayers, David Melka, Phillip Curry, **Justin Payne**, Anna Maounounen-Laasri, Charles Wang, Thomas Hammack, and Eric Brown. *Draft Genome Sequences of Salmonella enterica subsp. enterica Serovar Typhimurium and Serovar Nottingham Isolated from Food Products.* Genome Announc. (to appear)

83. Fernanda Almeida, Marta Inês Medeiros, Dália Rodrigues, **Justin Payne**, Ruth Timme, Marc Allard, and Juliana Falcão. *Draft Genome Sequences of 40 Salmonella Typhimurium isolated from humans and food in Brazil* Genome Announc. (to appear)

---

(VI) Seminars
-------------

53. Marc Allard, Ruth E. Timme, Peter Evans, Marc W. Allard, Errol Strain, Charles Wang, George Kastanis, **Justin Payne**, Christine Keys, Steve Musser, Eric Brown.  *Integration of NGS Desktop Sequencers to Build a Global Genomic Network for Pathogen Trace-back and Outbreak Detection.* College Park MD. June 2014.

57. Ruth E. Timme, Peter Evans, Marc W. Allard, Errol Strain, **Justin Payne**, Christine Keys, Steve Musser. *FDA GenomeTrakr: building an international public heath lab network for foodborne pathogen tracking.* Sequencing, Finishing, and Analysis in the Future, Santa Fe, NM May 2014.

3. Eric Stevens, **Justin Payne**, Bill Klimke, James Pettengill, Ruth E. Timme, Magaly Toro, Marc Allard, Andrea Ottesen, James Robert White, Karen Jarvis, Eija Trees, Heather Carleton. *FDA-CFSAN/JIFSAN-IFSTL Whole Genome Sequencing Pilot Course.* (participated as bioinformatics workshop instructor.) College Park MD. (workshop) March 2015.

4. Eric Stevens, **Justin Payne**, Bill Klimke, James Pettengill, Ruth E. Timme, Charles Wang, Karen Jarvis, Eija Trees, Heather Carleton. *FDA-CFSAN/JIFSAN-IFSTL Whole Genome Sequencing Course.* (participated as bioinformatics workshop instructor.) College Park MD. (workshop) Sept 2015.

(VII) Scientific Training
------------------------

1. **February 2012** - Next-gen Sequencing and Bioinformatics, Institute for Genome Sciences, University of Maryland, Baltimore
2. **June 2013** - PacBio User Group Meeting and Bioinformatics Workshop 2013, IGS, UMD Baltimore
3. **May 2014** - NGS Typing of Bacterial Pathogens, American Society for Microbiology General Meeting 2014, Boston
4. **June 2014** - PacBio User Group Meeting and Bioinformatics Workshop 2014, IGS, UMD Baltimore
5. **June 2015** - PacBio User Group Meeting and Bioinformatics Workshop 2015, IGS, UMD Baltimore

---

(VIII) Honors and Awards
-----------------------

* 2011 - Best Class Presentation, Protein Engineering, University of Nebraska, Lincoln
* 2014 - HHS Innovates Awards, Secretary's Pick for "Whole Genome Sequencing: Future of Food Safety"
* 2015 - NCBI Hackathon, "Genius Mole People who code in Java" recipient

---

(IX) Scientific Conferences and Meetings
------------------------------------------

8. **June 2016** *Sequencing, Finishing, and Analysis in the Future*, Santa Fe
7. **September 2015** *American Society for Microbiology, Next-Gen Sequencing Pipelines*, Washington DC
6. **September 2015** *Next Generation Sequencing Network for Food Pathogen Traceability Meeting #2*, Washington DC
5. **August 2015** - *NCBI Hackathon*, Bethesda
4. **June 2015** - *Sequencing, Finishing, and Analysis in the Future*, Santa Fe
1. **September 2014** - *Real-Time Genome Sequencing Surveillance Multi-Agency Collaboration 2014*, Washington DC
2. **April 2014** - *American Society for Microbiology General Meeting 2014*, Boston
3. **September 2013** - *Next Generation Sequencing Network for Food Pathogen Traceability Meeting #1*, Baltimore


(X) Projects
-------------
**Bootsie** - Bootstrap Analysis of RFLP : a Java-based biostatistics utility used by UNL and USDA-ARS entomologists to analyze population structure. Includes a suite of file-conversion utilities.  
**Sif** - Sequence Information Finder : a suite of command-line utilities for quick and easy lookups in the FDA-CFSAN GenomicsDB.  
**BaseCase** - Web-based Pipeline Management : an intuitive system for hooking up bioinformatics pipelines using powerful concepts from functional programming combined with process and filesystem isolation via Docker. Pipeline workflows can be elegantly and seamlessly shared between workgroups without the need to manually manage software installs or curate reference databases.  
**GenomeTrakr** - Real-time Surveillance of Food-borne Pathogens : nationwide system of networked Illumina sequencers for pathogen traceback and surveillance. Developed key integration-layer architecture.  
**Sra-quick-submit** - Rapid SRA submission generation for GenomeTrakr contributors. Currently in world-wide use.  
**GIMS** - Genomic Information Management System : a web-based LIMS and file-hosting repository combining rigorous lab activity tracking with rich sample tracking and semantic linking, with links to bioinformatics analysis systems and HPC.
**Watchos Grande** - Lightweight process-heartbeat tracking in a Flask/Gunicorn app.

---

(XI) References
--------------

* Dr. Marc W. Allard - FDA-CFSAN, College Park MD, (240) 402-2941, *marc.allard@fda.hhs.gov*
* Dr. Bruce E. Hibbard - USDA-ARS, Columbia MO, (573) 882-6281, *HibbardB@missouri.edu*
* Dr. John Foster - Uni. Nebraska-Lincoln NE, (402) 472-8686, *john.foster@unl.edu*
* Ben Follis - Senior Computer Scientist, Adobe Systems, *ben@follis.net*
