Justin Payne
============


> I'm a scientific toolmaker looking to solve interesting problems in genomic science.
> 
> Specialties: Integration, Full-stack Web, Bioinformatics Service Architectures.

---

Information
-----------
1155 Ripley St Apt 1814  
Silver Spring, MD 20910  

*Justin.Payne@fda.hhs.gov*  
+1 402.909.1436  

GitHub Username: [crashfrog](http://github.com/crashfrog)

---
 
Education
---------
Date             | Degree and Program
----------------:|:-----------
2009 - 2012      |Earned **Bachelor of Science in Biochemistry**, minors in Chemistry and Mathematics, University of Nebraska - Lincoln
1999 - 2003      |Studied for Bachelors of Arts in English and Computer Science, Gustavus Adolphus College (on hiatus)
Aug 1998         |Certificate of French proficiency, Université Paris-Sorbonne

 
---

Professional Experience
-----------------------
Date                | Organization and Duties
:------------------:|:-----------------------
Oct 2012 - today    |**FDA-CFSAN-ORS-DM-MMSB (ORAU Fellowship)**
                    |**ORISE Senior Bioinformatics Developer** supporting the analysis and interpretation of whole genome sequencing data from bacterial food-borne pathogens, and the development of bioinformatic pipelines and pipeline frameworks to aid in the detection, identification, and traceback of microbial food-borne pathogens: Trained in sample handling, preparation, and sequencing on IonTorrent PGM sequencing instruments. Demonstrated expertise in command-line operation of Linux and Mac OS X (desktop, server, and high-performance clusters.) Wrote high-performance analysis tools in C/C++, Python 2/3, Java 6, and JavaScript to analyze and interpret molecular data from DNA sequencing. Developed GenomicsDB database for isolate metadata, sequencing scheduling, and high-throughput submission to national biotechnology databases such as NCBI GenBank and NCBI SRA, and primary data architect in the project to migrate this system to Genhom's SLims Genomic Information Management System (GIMS). Wrote integration-layer packages in Python to automatically collect, assess, quality-control, assemble, and analyze genomic sequencing data from Illumina MiSeq/NextSeq, IonTorrent PGM, and PacBio RS sequencing instruments locally or connected to the FDA network via VPN connections or through Illumina BaseSpace cloud services. Primary author and maintainer of the FDA-hosted codebase and informatic infrastructure for FDA/State joint GenomeTrakr Network, FDA/CDC Real-Time Listeria Genomic Surveillance Project, and FDA-CFSAN/MN/WA/NY Real-Time Salmonella Enteriditis Project, resulting in the deposit of almost 10,000 bacterial genomes into publicly-accessible databases. Developed sra-quick-submit, an easy-to-use NCBI Sequence Read Archive (SRA) submission tool used by researchers around the world to submit their own genomes to the GenomeTrakr project. Utilized High-Performance Computing (HPC) and Field-Programmable Gate Array (FPGA) coprocessor platforms for high- throughput genomic science. Evaluated over 40 bioinformatics tools for pipeline incorporation and analysis use, determined their relevance to the food safety mission, and provided local support for their deployment and use. Provided bioinformatics system development consultation on an informal basis to researchers at diverse US and international agencies and groups. Developer of BaseCase, a web-based bioinformatics engine leveraging best-of-breed process and filesystem isolation tools like Docker to enable reproducible, sharable, versioned workflow pipelines with full auditing and data provenance.
                    |
Feb 2012 - Oct 2012 |**Trojan Defense, LLC**
                    |**Analysis Software Programmer** developing test harnesses for proprietary solid-state radiologic sensors in a PC-based visualization environment. Specialized in "bare-metal" C/C++ on Atmel-family embedded SoC (Systems on a Chip.) Developed custom evaluation boards based on Atmel ATMega 1280 CPU. Used professional illustration tools (Adobe family, Inkscape) to produce technical illustrations, patent documentation, and marketing materials. Wrote proposals for Small Business Innovation Research (SBIR) grants for DTRA, DoD, DoE, and other Federal agencies.
                    |
May 2005 - Dec 2006 |**University of Missouri-Columbia/USDA-ARS**
                    |**Biological Field Research Aide (GS-1)** assisting in field studies in agricultural entomology, jointly as an appointment to the USDA-ARS and as an employee of the University of Missouri, Columbia. Developed professional web presence and custom content management system (CMS) using Apache and PHP. Developed laboratory setup and tools for electroanntennaographic bioassay of _Diabrotica vergifera v._ (western corn rootworm) and _Loxosceles reclusa_ (brown recluse) species for detection of host identification factors. Constructed field study equipment using basic and power carpentry tools. Managed small (3-6 person) groups of fellow assistants. Worked independently to ensure adherence to study protocols.
                    |
Sept 1998 - Jan 2003|**Gustavus Adolphus College, Dept. of Information Technologies**
                    |**Tech Support Consultant** providing phone and site-based technical support on Windows 95/98/ME/2000/NT/XP, Mac OS 9/10, and Linux (Debian/Ubuntu/Fedora/Slackware) systems for students, staff, and faculty. Wrote training materials.
                    
---



Technologies
------------

**Platforms**

* Linux (Ubuntu/CentOS/RHEL/Arch)
* Mac OS X (desktop, server)
* Atmel family of embedded microprocessors
* Xilinx Spartan-6 FPGAs
* Amazon EC2

**Programming Languages**

* Python
* Perl
* PHP
* C/C++
* Java
* JavaScript
* HTML5+CSS3

**Databases**

* MySQL
* SQL Server 2008
* PostgreSQL
* Oracle
* Neo4J

**Development Frameworks** 

* Django
* Flask
* AWT/Swing
* Angular.js
* JSPlumb

---

Publications
------------

2. Marc Allard, Tim Muruvanda, Errol Strain, Ruth Timme, Yan Luo, Charles Wang, Christine Keys, **Justin Payne**, Tony Cooper, Khai Luong, Yi Song, Chen-Shan Chin, Jonas Korlach, Richard Roberts, Peter Evans, Steven Musser, and Eric Brown. *A fully assembled genome sequence for Salmonella. Javiana strain CFSAN001992.* Genome Announc. 2013 Mar 21;1(2):e0008113.

1. Narjol González-Escalona, Melinda McFarland, Lydia Rump, **Justin Payne**, Denis Andrzejewski, Eric Brown, Peter Evans, and Timothy Crowley. *Draft genome sequences of two O104:H21 Escherichia coli isolates causing hemorrhagic colitis during an outbreak in Montana in 1994 provide insight into their pathogenicity.* Genome Announc. 2013 Oct 3;1(5). pii: e00805-13.

3. Maria Hoffmann, Tim Muruvanda, Marc Allard, Jonas Korlach, Richard Roberts, Ruth Timme, **Justin Payne**, Patrick McDermott, Peter Evans, Jianghong Meng, Eric Brown, and Shaohua Zhao. *Complete Genome Sequence of a Multidrug-Resistant Salmonella enterica Serovar Typhimurium var. 5- Strain Isolated from Chicken Breast.* Genome Announc. 2014 Apr 3;2(2). pii: e00294-14

4. Julie Haendiges, Ruth Timme, Marc Allard, Robert Myers, **Justin Payne**, Eric Brown, Peter Evans, and Narjol González-Escalona. *Draft genomes of clinical Vibrio parahaemolyticus strains isolated in Maryland (2010-2013).* Genome Announc. 2014 Aug 7;2(4). pii: e00776-14.

5. Maria Hoffmann, Tim Muruvanda, Cary Pirone, Jonas Korlach, Ruth Timme, **Justin Payne**, Peter Evans, Jianghong Meng, Eric Brown, and Marc Allard. *First Fully Assembled Genome Sequence of Salmonella enterica Serovar Cubana associated with a Multistate Outbreak in the United States.* Genome Announc. 2014 *(to appear)*


---

Projects
--------
**Bootsie** - Bootstrap Analysis of RFLP : a Java-based biostatistics utility used by UNL and USDA-ARS entomologists to analyze population structure. Includes a suite of file-conversion utilities.  
**Sif** - Sequence Information Finder : a suite of command-line utilities for quick and easy lookups in the FDA-CFSAN GenomicsDB.  
**BaseCase** - Web-based Pipeline Management : an intuitive system for hooking up bioinformatics pipelines using powerful concepts from functional programming combined with process and filesystem isolation via Docker. Pipeline workflows can be elegantly and seamlessly shared between workgroups without the need to manually manage software installs or curate reference databases.  
**GenomeTrakr** - Real-time Surveillance of Food-borne Pathogens : nationwide system of networked Illumina sequencers for pathogen traceback and surveillance. Developed key integration-layer architecture.   
**Sra-quick-submit** - Rapid SRA submission generation for GenomeTrakr contributors. Currently in world-wide use.

---

References
----------

* Dr. Marc W. Allard - FDA-CFSAN, College Park MD, (240) 402-2941
* Dr. Bruce E. Hibbard - USDA-ARS, Columbia MO, (573) 882-6281
* Dr. John Foster - Uni. Nebraska-Lincoln NE, (402) 472-8686
